src/platform/android/java/chip/platform/DiagnosticDataProviderImpl.java

```diff
@@ -0,0 +1,107 @@
+/*
+ *   Copyright (c) 2022 Project CHIP Authors
+ *   All rights reserved.
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License");
+ *   you may not use this file except in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ *
+ */
+package chip.platform;
+
+import android.content.Context;
+import android.util.Log;
+import java.net.Inet4Address;
+import java.net.Inet6Address;
+import java.net.InetAddress;
+import java.net.SocketException;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Enumeration;
+import java.util.List;
+
+public class DiagnosticDataProviderImpl implements DiagnosticDataProvider {
+  private static final String TAG = DiagnosticDataProviderImpl.class.getSimpleName();
+  private Context mContext;
+
+  public DiagnosticDataProviderImpl(Context context) {
+    mContext = context;
+  }
+
+  @Override
+  public int getRebootCount() {
+    // TODO: boot count is not supported until api 24, please do it at factory app
+    //        try {
+    //            return Settings.Global.getInt(mContext.getContentResolver(),
+    // Settings.Global.BOOT_COUNT);
+    //        } catch (Settings.SettingNotFoundException e) {
+    //            Log.e(TAG, "getRebootCount: " + e.getMessage());
+    //        }
```

Comment:
You should be able do this:
```
if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.N) {
  // BOOT_COUNT code
} else {
  // return default value
}
```
You also may need some annotation like `@SuppressLint`, not entirely sure there.

Reply:
I did not find any easy to upgrade the API level in gn

Reply:
I seem to remember this working even at build-time as long as you use an annotation...but I won't block.